raco/backends/myria/myria.py

```diff
@@ -1914,14 +1902,44 @@ def fire(self, op):
 
 class FillInJoinPullOrder(rules.Rule):
 
-    def __init__(self, pull_order):
+    def __init__(self):
         self._disabled = False
-        self.pull_order = pull_order
+
+    def get_idb_side(self, op):
```

Comment:
Turned out that `preorder` doesn't work if `stop_recursion` is added to it, since finding an `MyriaIDBController` requires to pass a consumer with `stop_recursion=True`.